lib/active_record_shards/configuration_parser.rb

```diff
@@ -29,10 +29,13 @@ def explode(conf)
           conf["#{env_name}_replica"] = replica_conf
         end
 
-        if replica_conf = env_config.delete('slave')
+        # rubocop:disable Style/Next
+        if legacy_replica_conf = env_config.delete('slave')
           ActiveRecordShards::Deprecation.warn('`slave` configuration keys should be replaced with `replica` keys!')
-          env_config['replica'] ||= replica_conf
+          expand_child!(env_config, legacy_replica_conf)
```

Comment:
❤️ Of course.

This 2nd `if` block was previously places above the 1st `if` block, and then of course it would have worked fine.